packages/editor/src/editor/browser/cms/addCmsQuickAction-utils.ts

```diff
@@ -0,0 +1,44 @@
+export type SavedSelection = { start: number; end: number };
+export type InputTextAreaRef = React.RefObject<HTMLInputElement> | React.RefObject<HTMLTextAreaElement>;
+
+export const handleTextSelection = (ref: InputTextAreaRef, setSavedSelection: (value: SavedSelection | undefined) => void) => {
+  if (ref.current) {
+    const selectionStart = ref.current.selectionStart ?? 0;
+    const selectionEnd = ref.current.selectionEnd ?? 0;
+    if (selectionStart !== selectionEnd) {
+      setSavedSelection({ start: selectionStart, end: selectionEnd });
+    } else {
+      setSavedSelection(undefined);
+    }
+  }
+};
+
+export const getSelectedText = (ref: InputTextAreaRef, savedSelection: SavedSelection | undefined) => {
+  if (ref.current && savedSelection) {
+    const text = ref.current.value.substring(savedSelection.start, savedSelection.end);
+    return text;
+  }
+  return '';
+};
+
+export const canPopoverBeRendered = (ref: InputTextAreaRef, savedSelection: SavedSelection | undefined) => {
```

Comment:
strange name to me. If I read this I could get the idea that it is sometimes not possible to render the popover?

You control if the quick fix buttons are displayed right?
So then we should call it this way: `showQuickFix` or `quickFixVisible`

By the way the term `QuickAction` is already taken by the `QuickActionBar`. So I would not mixup this here. I would suppose to rename this feature to `QuickFix`

Reply:
So should i also rename the metas?
meta/cms/cmsQuickActions --> meta/cms/cmsQuickFixes
meta/cms/executeCmsQuickAction --> meta/cms/executeCmsQuickFix

Reply:
honestly, I think it would make sense yes...
But at least I think you should rename the frontend stuff

Reply:
All right, I'll change everything from Action to Fix, except the metas for now. I will also rename the metas in a later PR.